app/controllers/thredded/posts_controller.rb

```diff
@@ -37,8 +37,22 @@ def destroy
                     notice: I18n.t('thredded.posts.deleted_notice')
     end
 
+    def mark_as_unread
+      authorize post, :read?
+      post.mark_as_unread(thredded_current_user, post.page(user: thredded_current_user))
```

Comment:
This does not work for private messages (I'm guessing the `page` method needs to be updated)